backend/mem/fileSystem.go

```diff
@@ -0,0 +1,176 @@
+package mem
+
+import (
+	"path"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
+type obj struct {
+	isFile bool
+	i      interface{}
+}
+type objMap map[string]*obj
+type FileSystem struct {
+	fsMap map[string]objMap
+}
+
+// FileSystem will return a retrier provided via options, or a no-op if none is provided.
+func (fs *FileSystem) Retry() vfs.Retry {
+	return vfs.DefaultRetryer()
+}
+
+/*
+NewFile function returns the mem implementation of vfs.File.
+Since this is inside "fileSystem" we assume that the caller knows that the CWD is the root.
+If a non-absolute path is given, an error is thrown.Additionally, a file does not
+technically exist until a call to "Touch()" is made on it. The "Touch" call links the
+file with the fileSystem's map and brings it into existence.
+If a file is written to before a touch call, Write() will take care of that call.  This is
+true for other functions as well and existence only poses a problem in the context of deletion
+or copying FROM a non-existent file.
+*/
+func (fs *FileSystem) NewFile(volume string, absFilePath string) (vfs.File, error) {
+
+	err := utils.ValidateAbsoluteFilePath(absFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	file, err := newFile(path.Base(absFilePath)) //validateAbsFile path will throw an error if there was a trailing slash, hence not calling path.Clean()
+	if err != nil {
+		return nil, err
+	}
+	tmp, err := fs.NewLocation(volume, utils.EnsureTrailingSlash(path.Dir(absFilePath)))
+	if err != nil {
+		return nil, err
+	}
+
+	file.location = tmp
+	return file, nil
+}
+
+/*
+NewLocation function returns the mem implementation of vfs.Location.
+A location does not exist unless at least one file lives on it. If a file
+is created on a non-existent location, then it will be created
+*/
+func (fs *FileSystem) NewLocation(volume string, absLocPath string) (vfs.Location, error) {
+
+	err := utils.ValidateAbsoluteLocationPath(absLocPath)
+	if err != nil {
+		return nil, err
+	}
+	str := utils.EnsureTrailingSlash(path.Clean(absLocPath))
+	return &Location{
+		fileSystem: fs,
+		name:       str,
+		exists:     false,
+		volume:     volume,
+	}, nil
+
+}
+
+// Name returns "mem"
```

Comment:
Something along the lines of:
// Name returns the name of the file system.
would be more accurate and prevent you from needing to update the comment if the name constant ever gets updated. IE: it doesn't actually return "mem" it returns "In-Memory Filesystem" as defined currently.